mobile/types.go

```diff
@@ -279,8 +279,8 @@ func (tx *Transaction) GetTo() *Address {
 	return nil
 }
 
-func (tx *Transaction) WithSignature(sig []byte) (signedTx *Transaction, _ error) {
-	rawTx, err := tx.tx.WithSignature(types.HomesteadSigner{}, sig)
+func (tx *Transaction) WithSignature(sig []byte, chainID *BigInt) (signedTx *Transaction, _ error) {
+	rawTx, err := tx.tx.WithSignature(types.NewEIP155Signer(chainID.bigint), sig)
```

Comment:
If I pass in `null` for the chainID (allowed for `GetFrom()`), it will panic. Would be nice to use the homestead signer for that.

My suggestion would be:

```go
func (tx *Transaction) WithSignature(sig []byte, chainID *BigInt) (signedTx *Transaction, _ error) {
    signer := types.HomesteadSigner{}
    if chainID != nil {
        signer = types.NewEIP155Signer(chainID.bigint)
    }
    rawTx, err := tx.tx.WithSignature(signer, sig)
    return &Transaction{rawTx}, err
}
```

Could you please also update `GetFrom()` so that it uses the `HomesteadSigner` if null is passed in as above, instead of defaulting to the 0 chain id? Apparently the old solution is wrong, just happens to work because EIP155 is backward compatible.